src/txmempool.cpp

```diff
@@ -97,6 +99,28 @@ void CTxMemPool::UpdateForDescendants(txiter updateIt, cacheMap &cachedDescendan
     mapTx.modify(updateIt, update_descendant_state(modifySize, modifyFee, modifyCount));
 }
 
+void CTxMemPool::GetRebroadcastTransactions(std::set<uint256>& setRebroadcastTxs)
+{
+    // Don't rebroadcast txns during importing, reindex, or IBD to ensure we don't
+    // accidentally spam our peers with old transactions.
+    if (::ChainstateActive().IsInitialBlockDownload() || ::fImporting || ::fReindex) return;
+
+    BlockAssembler::Options options;
+    options.nBlockMaxWeight = MAX_REBROADCAST_WEIGHT;
+    CScript scriptDummy = CScript() << OP_TRUE;
+
+    // use CreateNewBlock to get set of transaction candidates
+    std::unique_ptr<CBlockTemplate> pblocktemplate = BlockAssembler(Params(), options).CreateNewBlock(scriptDummy);
+
+    LOCK(cs);
+    for (const auto& tx : pblocktemplate->block.vtx) {
+        if (mapTx.find(tx->GetHash()) == mapTx.end()) continue;
```

Comment:
ok my understanding is.. without the check, if the txn was no longer in the mempool, it would get added to `setRebroadcastTxs` here. Returned to the caller here: https://github.com/bitcoin/bitcoin/pull/16698/files#diff-eff7adeaec73a769788bb78858815c91R3799


https://github.com/bitcoin/bitcoin/blob/master/src/net_processing.cpp#L3854

Reply:
ok my understanding is.. without the check, if the txn was no longer in the mempool, it would..
1.  get added to setRebroadcastTxs in this function. 
2. returned to the caller [here](https://github.com/bitcoin/bitcoin/pull/16698/files#diff-eff7adeaec73a769788bb78858815c91R3799) & insert into `setInventoryTxToSend`
3. `setInventoryTxToSend` gets copied into `vInvTx` [here](https://github.com/bitcoin/bitcoin/blob/master/src/net_processing.cpp#L3825)
4. checks if txn is in mempool otherwise skips it [here](https://github.com/bitcoin/bitcoin/blob/master/src/net_processing.cpp#L3854) 

which all seems fine so I'll remove the check.